packages/alert/src/Alert.tsx

```diff
@@ -49,9 +52,48 @@ export interface AlertProps extends React.HTMLAttributes<HTMLElement> {
   /**
    * The variant to use. Defaults to "relative".
    */
-  type?: 'relative' | 'toast' | 'banner'
+  type?: AlertTypes
+
+  /**
+   * Show / Hide close button
+   */
+  showCloseButton?: boolean
+
+  /**
+   * Show / Hide Alert
+   */
+  isOpen?: boolean
+
+  /**
+   * Show alert duration
+   */
+  delay?: number
+
+  /**
+   * Offset Index
+   */
+  stackIndex?: number
 }
 
+type TypeProps =
+  | {
+      type: AlertTypes.TOAST
+      delay?: number
+      showCloseButton: false
+      onClose: () => void
+    }
+  | {
+      type?: AlertTypes.RELATIVE | AlertTypes.BANNER
```

Comment:
Here`type` should not be optional here; what should be optional is `onClose`, Is there a typescript error that you are trying to bypass by doing this?